packages/salesforcedx-vscode-apex/src/views/testOutline.ts

```diff
@@ -0,0 +1,465 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as os from 'os';
+import * as vscode from 'vscode';
+import events = require('events');
+import fs = require('fs');
+import ospath = require('path');
+import {
+  CliCommandExecutor,
+  Command
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { getApexTests, isLanguageClientReady } from '..';
+import {
+  APEX_GROUP_RANGE,
+  DARK_BLUE_BUTTON,
+  DARK_GREEN_BUTTON,
+  DARK_RED_BUTTON,
+  LIGHT_BLUE_BUTTON,
+  LIGHT_GREEN_BUTTON,
+  LIGHT_RED_BUTTON
+} from '../constants';
+import { nls } from '../messages';
+import { ApexTestMethod } from './LSPConverter';
+import { FullTestResult, TestSummarizer } from './TestDataAccessObjects';
+
+// Message
+const LOADING_MESSAGE = nls.localize('force_test_view_loading_message');
+const NO_TESTS_MESSAGE = nls.localize('force_test_view_no_tests_message');
+const NO_TESTS_DESCRIPTION = nls.localize(
+  'force_test_view_no_tests_description'
+);
+
+// Import from core
+const sfdxCoreExports = vscode.extensions.getExtension(
+  'salesforce.salesforcedx-vscode-core'
+)!.exports;
+const EmptyParametersGatherer = sfdxCoreExports.EmptyParametersGatherer;
+const SfdxCommandlet = sfdxCoreExports.SfdxCommandlet;
+const ForceApexTestRunCodeActionExecutor =
+  sfdxCoreExports.ForceApexTestRunCodeActionExecutor;
+const SfdxWorkspaceChecker = sfdxCoreExports.SfdxWorkspaceChecker;
+const channelService = sfdxCoreExports.channelService;
+
+export class ApexTestOutlineProvider
+  implements vscode.TreeDataProvider<TestNode> {
+  private onDidChangeTestData: vscode.EventEmitter<
+    TestNode | undefined
+  > = new vscode.EventEmitter<TestNode | undefined>();
+  public onDidChangeTreeData = this.onDidChangeTestData.event;
+
+  private eventsEmitter = new events.EventEmitter();
+
+  private apexTestMap: Map<string, TestNode> = new Map<string, TestNode>();
+  private rootNode: TestNode | null;
+  private static testStrings: Set<string> = new Set<string>();
+  private path: string;
+  private apexTestInfo: ApexTestMethod[] | null;
+
+  constructor(path: string, apexTestInfo: ApexTestMethod[] | null) {
+    this.rootNode = null;
+    this.path = path;
+    this.apexTestInfo = apexTestInfo;
+    this.getAllApexTests(this.path);
+    this.eventsEmitter.on('Show Highlight', this.updateSelection);
```

Comment:
@jkalloor0701 - Let's use a safer type of event name to avoid collision. Use something like `sfdx:highlight_test`. I'm actually not sure why you called this highlight. Is "highlight" the right description?